src/hotspot/share/gc/shenandoah/shenandoahRegulatorThread.hpp

```diff
@@ -65,16 +65,26 @@ class ShenandoahRegulatorThread: public ConcurrentGCThread {
   void stop_service();
 
  private:
-  void regulate_interleaved_cycles();
-  void regulate_concurrent_cycles();
-  void regulate_heap();
-
+  // When mode is generational
+  void regulate_young_and_old_cycles();
+  // When mode is generational, but ShenandoahAllowOldMarkingPreemption is false
+  void regulate_young_and_global_cycles();
+  // Default behavior for other modes (single generation).
+  void regulate_global_cycles();
+
+  // These return true if a cycle was started.
   bool start_old_cycle();
   bool start_young_cycle();
   bool start_global_cycle();
 
   bool should_unload_classes();
```

Comment:
Could you also please write a 1-line documentation comment for this as well?

Reply:
Done.